bridge/tfchain_bridge/pkg/bridge/withdraw.go

```diff
@@ -70,12 +70,51 @@ func (bridge *Bridge) handleWithdrawCreated(ctx context.Context, withdraw subpkg
 func (bridge *Bridge) handleWithdrawExpired(ctx context.Context, withdrawExpired subpkg.WithdrawExpiredEvent) error {
 	logger := log.Logger.With().Str("trace_id", fmt.Sprint(withdrawExpired.ID)).Logger()
 
-	ok, source := withdrawExpired.Source.Unwrap()
+	ok, source := withdrawExpired.Source.Unwrap() // transfers from the previous runtime before 147 has no source address
 
 	if !ok {
-		// log and skip ? FATAL ?
-		logger.Error().Msg("this should never be triggered unless this bridge release was deployed before the runtime 147 upgrade!")
-		return nil
+		// This path is intended solely for processing transfers that lack a source address
+		// and should be retained until the network has been verified to have no transfers from the previous runtime before 147.
+
+		if err := bridge.wallet.CheckAccount(withdrawExpired.Target); err != nil {
+			logger.Warn().
+				Str("event_action", "transfer_failed").
+				Str("event_kind", "alert").
+				Str("category", "transfer").
+				Dict("metadata", zerolog.Dict().
+					Str("reason", err.Error())).
+					Str("type", "burn").
+				Msg("a withdraw failed with no way to refund!") 
+			return bridge.subClient.RetrySetWithdrawExecuted(ctx, withdrawExpired.ID)
+		}
+
```

Comment:
In this case what I understand is that, before 147, `Source` is not provided and `Target` do not exist (never existed or not existing anymore) so funds are stuck in the bridge because nowhere to transfer it, correct?

Reply:
yes. Also account can be exists but have no tft trustline. in this case also you can't either bridge or refund the transfer prior 147.